packages/calcite-components/src/tests/commonTests/cancelableResource.ts

```diff
@@ -0,0 +1,35 @@
+import { mount } from "@arcgis/lumina-compiler/testing";
+import { describe, expect, it, vi } from "vitest";
+import { mockConsole } from "../../tests/utils/logging";
+
+/**
+ * Helper for testing cancelable behavior in components (debounced or throttled).
+ *
+ * @example
+ * describe("cancelable behavior", () => {
+ *   cancelableBehavior("<calcite-action-bar>", "resize"");
+ *   cancelableBehavior("<calcite-color-picker>", "drawColorControls");
+ * });
+ *
+ * @param {ComponentTag} componentTag - The tag name of the component to test.
+ * @param {string} cancelableMethod - The name of the debounced or throttled method to test.
+ */
+export function cancelableBehavior(componentTag: string, cancelableMethod: string): void {
```

Comment:
I’m a bit concerned this API requires implementation knowledge. Would it be simpler for the util to verify that at least one cancelable is added, and that all of their cancel methods are called on disconnect? We could enforce usage of the controller via a linting (e.g., block anything using `cancel` in favor of the util). I think we could do this via `no-restricted-properties` or `no-restricted-syntax`. It won't catch anything that's not being canceled though, although this is possible regardless of approach.

@Elijbet @driskull WDYT?

Reply:
Yeah that sounds good to me